code/game/machinery/bots/cprbot.dm

```diff
@@ -0,0 +1,491 @@
+		if (!isnull(other_cpr_bot.human))
+			other_bot_patient = other_cpr_bot.human.resolve()
+
+		if (!isnull(other_bot_patient) && (other_bot_patient in potential_patients))
+			potential_patients.Remove(other_bot_patient) // Remove the patient targeted by another bot
+
+	// If potential patients are found, target the first one
+	if (length(potential_patients))
+		var/mob/living/carbon/human/patient = potential_patients[1]
+		human = WEAKREF(patient)
+
+		if (patient && !(patient in has_said_to_patient))
+			visible_message("[patient] is injured! I'm coming!")
+			has_said_to_patient += patient
+
+		move_to_target()
+	else
+		// If no patients are found, follow the owner if idle
+		if (state == STATE_CPRBOT_IDLE && owner && (owner in view(search_radius, src)))
+			state = STATE_CPRBOT_FOLLOWING_OWNER
+			walk_to(src, owner, 0, movement_delay)
+		else if (state == STATE_CPRBOT_FOLLOWING_OWNER)
+			// Continue following the owner
+			walk_to(src, owner, 0, movement_delay)
+		else
+			// Default to idle state
+			go_idle()
+
+/obj/structure/machinery/bot/cprbot/proc/call_astar_pathfinding()
+	var/mob/living/carbon/human/patient = human.resolve()
+	if (isnull(patient))
+		return FALSE
+
+	var/turf/target_turf = get_turf(patient)
+
+	if (!target_turf)
+		return
+
+	// Calculate the path using your A* function
+	path = AStar(loc, target_turf, /turf/proc/CardinalTurfsWithAccess, /turf/proc/Distance, 0, 30, id=botcard)
+
+	if (!length(path))
+		return
+
+	// Delete the straight line sections since walk_to moves better without that.
+	var/list/path_new = list()
+	var/turf/last = path[length(path)]
+
+	if (!istype(last, /turf)) // Ensure last is a turf before accessing it
+		return
+
+	path_new.Add(path[1])
+
+	for (var/i = 2; i < length(path); i++)
+		if (istype(path[i], /turf) && istype(path[i + 1], /turf))
+			var/turf/current_turf = path[i]
+			var/turf/next_turf = path[i + 1]
+
+			if ((next_turf.x == current_turf.x) || (next_turf.y == current_turf.y)) // We have a straight line, scan for more to cut down
+				path_new.Add(current_turf)
+				for (var/j = i + 1; j < path.len; j++)
+					if (istype(path[j + 1], /turf))
+						var/turf/next_next_turf = path[j + 1]
+						var/turf/prev_turf = path[j - 1]
+
+						// This is a corner and the endpoint of our line
+						if ((next_next_turf.x != prev_turf.x) && (next_next_turf.y != prev_turf.y))
+							path_new.Add(path[j])
+							i = j + 1
+							break
+
+
+
+					if (j == path.len - 1)
+						path = list()
+						path = path_new.Copy()
+						path.Add(last)
+						return
+
+
+			else
+				path_new.Add(current_turf)
+
+	path = list()
+	path = path_new.Copy()
+	path.Add(last)
+
+	if (!path || length(path) == 0)
+		human = null
+		currently_healing = FALSE
+		return
+
+	// Start moving
+	state = STATE_CPRBOT_MOVING
+	move_to_target()
+	return
+
+/obj/structure/machinery/bot/cprbot/proc/can_still_see_patient()
+	if (isnull(human))
+		return FALSE
+
+	var/mob/living/carbon/human/patient = human.resolve()
```

Comment:
```suggestion
	var/mob/living/carbon/human/patient = human?.resolve()
```

Reply:
?. operator is same as `.` but if the variable before `?.` is null then the access attempt (here resolve()) will be canceled, unlike just `.`, which will try to access it anyways, and runtime.